core/controllers/admin_test.py

```diff
@@ -2724,6 +2725,162 @@ def test_handler_raises_error_with_invalid_exploration_version(
+            'exploration_id': self.EXP_ID_7,
+            'status': 'Draft',
+            'planned_publication_date_msecs': 100,
+            'last_modified_msecs': 100,
+            'first_publication_date_msecs': 200,
+            'unpublishing_reason': None
+        }
+        story.story_contents.nodes = [
+            story_domain.StoryNode.from_dict(self.node_1),
+            story_domain.StoryNode.from_dict(self.node_2),
+            story_domain.StoryNode.from_dict(self.node_3)
+        ]
+        self.nodes = story.story_contents.nodes
+        story.story_contents.initial_node_id = 'node_2'
+        story.story_contents.next_node_id = 'node_4'
+        story_services.save_new_story(self.admin_id, story)
+        self.subtopic_1 = topic_domain.Subtopic.create_default_subtopic(
+            1, 'Subtopic Title 1', 'sub-one-frag')
+        self.subtopic_2 = topic_domain.Subtopic.create_default_subtopic(
+            2, 'Subtopic Title 2', 'sub-two-frag')
+        self.SKILL_ID_1 = skill_services.get_new_skill_id()
+        self.SKILL_ID_2 = skill_services.get_new_skill_id()
+        self.subtopic_1.skill_ids = [self.SKILL_ID_1]
+        self.subtopic_2.skill_ids = [self.SKILL_ID_2]
+        self.save_new_topic(
+            self.TOPIC_ID_1, 'user', name='Topic',
+            description='A new topic', canonical_story_ids=[story.id],
+            additional_story_ids=[], uncategorized_skill_ids=[],
+            subtopics=[self.subtopic_1, self.subtopic_2], next_subtopic_id=3)
+        topic_services.publish_topic(self.TOPIC_ID_1, self.admin_id)
+        topic_services.publish_story(
+            self.TOPIC_ID_1, self.STORY_ID_1, self.admin_id)
+        story = story_fetchers.get_story_by_id(self.STORY_ID_1, strict=False)
+        chapters_change_list = []
+        for node in story.story_contents.nodes:
+            chapters_change_list.append(story_domain.StoryChange({
+                'cmd': story_domain.CMD_UPDATE_STORY_NODE_PROPERTY,
+                'node_id': node.id,
+                'property_name': (
+                    story_domain.STORY_NODE_PROPERTY_STATUS),
+                'old_value': node.status,
+                'new_value': constants.STORY_NODE_STATUS_DRAFT
+            }))
+        topic_services.update_story_and_topic_summary(
+            self.admin_id,self.STORY_ID_1 , chapters_change_list,
+                'Changes chapter status to draft', self.TOPIC_ID_1)
+        story = story_fetchers.get_story_by_id(self.STORY_ID_1, strict=False)
+
+        csrf_token = self.get_new_csrf_token()
+        generated_exps_response = self.post_json(
+            '/adminhandler', {
+                'action': 'publish_chapters_of_length_and_measurement_topic'
+            },
+            csrf_token=csrf_token)
+        story = story_fetchers.get_story_by_id(self.STORY_ID_1, strict=False)
+        for node in story.story_contents.nodes:
+            self.assertEqual(node.status, constants.STORY_NODE_STATUS_PUBLISHED)
```

Comment:
Add a precheck for this just before you do the csrf token and post json. The precheck should be formatted similarly to this and the only difference should be that the assertEqual expected value is different.

Reply:
Done!